src/pip/_internal/vcs/bazaar.py

```diff
@@ -48,29 +54,31 @@ def export(self, location, url):
 
         url, rev_options = self.get_url_rev_options(url)
         self.run_command(
-            ['export', location, url] + rev_options.to_args(),
+            make_command('export', location, url) + rev_options.to_args(),
```

Comment:
What if we implement `__iter__` for `RevOptions`, then turn all these into e.g. `make_command('export', location, url, *rev_options)`? It'll look even better when there's several options after `rev_options` that can just be included in the argument list to `make_command`.

Reply:
Thank you! 😊And great feedback!

> What if we implement `__iter__` for `RevOptions`

This is a really nice idea, except the only problem is that Python 2 doesn't allow positional arguments after an argument preceded by `*`:

```python
>>> foo(1, *[1, 2], 3)
  File "<stdin>", line 1
SyntaxError: only named arguments may follow *expression
```

What about changing `make_command()` to accept arguments with type `CommandArgs`? And then it could expand those internally. That would provide _almost_ the same convenience for the caller. The only downside is that it complicates `make_command()` slightly.

Reply:
Oh, dang.

I think that `make_command()` could handle it without too much complication, I would go for it. If we do it based on whether each argument is `Iterable` then we can still augment `RevOptions` and avoid having to put `to_args()` in these several places.

Reply:
> If we do it based on whether each argument is Iterable

I was thinking about this, too, but my concern was that if we make the annotation _too_ lenient, then we could accidentally be allowing other things, too. For example, while this isn't quite the same, an `str` object is also iterable, and if we weren't already allowing str, then it could e.g. lead to us inadvertently expand things like `'abc'` to `['a', 'b', 'c']`.

If we wanted to get fancy though, in the future we could maybe make our own duck-typing protocol that isn't already used for something else.